File: Programmers/Level2/다음큰숫자.py
def solution(n):
    answer = 0
    flag = False
    def getCntOne(n):
        cnt = 0
        binary = bin(n)[2:]
        for i in range(len(binary)):
            if binary[i] == '1':
                cnt += 1
        return cnt
    
    one_cnt = getCntOne(n)
    print(one_cnt)

    while 1:
        n += 1

        if getCntOne(n) == one_cnt:
            flag = True
        
        if flag:
            answer = n
            break
        
    return answer


n = 78
print(solution(n))

File: Programmers/Level2/다음큰숫자/다음큰숫자.js
function solution(n) {
  let answer = n + 1;
  const bin = n.toString(2);
  const cnt = bin.split("").filter((ele) => ele === "1").length;
  while (true) {
    if (
      cnt ===
      answer
        .toString(2)
        .split("")
        .filter((e) => e === "1").length
    ) {
      return answer;
    }
    answer += 1;
    if (answer > 1000000) return -1;
  }
}

const n = 78;
console.log(solution(n));
